packages/dev/core/src/Audio/v2/abstractAudioEngine.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable babylonjs/available */
+/* eslint-disable jsdoc/require-jsdoc */
+
+import type { AbstractAudioDevice } from "./abstractAudioDevice";
+import type { AbstractAudioListener } from "./abstractAudioListener";
+import type { AbstractAudioNode } from "./abstractAudioNode";
+import type { AbstractAudioPositioner } from "./abstractAudioPositioner";
+import type { AbstractAudioSender } from "./abstractAudioSender";
+import type { AbstractMainAudioBus } from "./abstractMainAudioBus";
+import type { AbstractStaticSoundInstance } from "./abstractStaticSoundInstance";
+import type { AbstractStaticSoundSource } from "./abstractStaticSoundSource";
+import type { AbstractStreamingSoundInstance } from "./abstractStreamingSoundInstance";
+import type { AbstractStreamingSoundSource } from "./abstractStreamingSoundSource";
+import type { IAudioNodeParent } from "./IAudioNodeParent";
+
+export abstract class AbstractAudioEngine implements IAudioNodeParent {
+    public dispose(): void {
+        for (const node of this._childNodes) {
+            node.dispose();
+        }
+        this._childNodes.length = 0;
+    }
+
+    private _childNodes = new Array<AbstractAudioNode>();
+
+    public _addChildNode(node: AbstractAudioNode): void {
```

Comment:
What would you do with the currently public `AbstractAudioEngine._addSoundInstance`, which will be called from the `AbstractSoundInstance` constructor?

I'm not sure this is a problem since we can just mark it as `@internal`. I see this being done a fair amount already in other classes. AFAIK the compiled JavaScript doesn't have private members anyway, and users are advised to avoid using members with an `_` prefix.

Reply:
Yes, properties/methods that we don't want the user to see but that we want to be able to use internally should be prefixed with `_`, but they should also be marked with `@internal` so that they don't appear in editors when using Intellisense.